packages/app/src/debug/empty/DebugEmptyView.vue

```diff
@@ -71,4 +158,80 @@ const loadingRows = [
   ['w-40px', 'w-[65%]'],
 ]
 
+const step = ref<number>()
+const lastStep = 3 // Treat the default state as the last slide
```

Comment:
A bit of a shame we've got some duplication here (`lastStep` should really be something like `slideshowimages.length`). I played with this part a bit but couldn't really find a nice abstraction to avoid this constant, though.

I feel like, ideally, `<Slideshow>` should actually receive and array of `steps`, and it might have an `onCompleted` props that would call the `slideshowCompleteMutation.executeMutation` code when the slideshow completes.

I'm not sure if we plan to have more slideshows, but if we do, it would be a little difficult to reuse this Slideshow component. Not sure on your bandwidth for the rest of the the sprint, but reconsidering the `<Slideshow>` component and making it a bit more reusable might be worth it? WDYT?

Reply:
I was considering the `steps` approach as well. The downside is the dynamic `<component :is` usage (not really a downside I just prefer being able to put the actual components in the template). I messed around with some advanced slot usage but didn't like how it was shaking out.

Check out dc9971caf79e9b1e95b804571a02401b11bcaa80, I pivoted to the steps approach and the complete emitter. LMK what you think!